blog/_posts/2019-08-01-artifacts.md

```diff
@@ -0,0 +1,179 @@
+---
+layout: post
+title: Pkg + BinaryBuilder -- The Next Generation
+author: Elliot Saba, Stefan Karpinski, Kristoffer Carlsson
+---
+
+# Pkg + BinaryBuilder -- The Next Generation
+
+Over the past few months, we have been iterating on and refining a design for `Pkg` to natively be able to reason about binary objects that are not Julia packages.  While the motivating application for this work has been improving the installation experience for binaries built with [`BinaryBuilder.jl`](https://github.com/JuliaPackaging/BinaryBuilder.jl), the artifacts subsystem is much more general.
+
+# Pkg Artifacts
+
+Pkg artifacts, as outlined in [`Pkg.jl#1234`](https://github.com/JuliaLang/Pkg.jl/issues/1234), provides a convenient way to associate containers of data with Julia projects and packages.  Artifacts are referred to by content-hash, or optionally by a name that is bound to a hash through an `Artifacts.toml` file.  An example `Artifacts.toml` file is shown here as an example:
+
+```TOML
+[socrates]
```

Comment:
Is there a reason this one is in single-brackets? Or just a typo?

Reply:
No typo, that's how TOML syntax works: a title in single brackets introduces a single-table entry, a title in double brackets appends a table to an array of tables. There is only one `socrates` artifact stanza which applies on all platforms, there are multiple `c_simple` artifact stanzas which apply on different platforms.